tools/generators/legacy/src/Generator/DisambiguateTargets.swift

```diff
@@ -32,26 +32,31 @@ extension Generator {
             [NameAndProductType: Set<String>] = [:]
         var names: [String: TargetComponents] = [:]
         var labels: [String: TargetComponents] = [:]
-        for (key, target) in targets {
-            let normalizedName = target.normalizedName
-            let normalizedLabel = target.normalizedLabel
+        for target in targets.values {
             labelsByNameAndProductType[.init(target: target), default: []]
-                .insert(normalizedLabel)
-            names[normalizedName, default: .init()]
-                .add(target: target, key: key)
-            labels[normalizedLabel, default: .init()]
-                .add(target: target, key: key)
+                .insert(target.normalizedLabel)
+        }
+        for (key, target) in targets {
+            let nameAndProductType = NameAndProductType(target: target)
+            if labelsByNameAndProductType[nameAndProductType]!.count == 1 {
```

Comment:
We know this force unwrap is safe since 

`targets` not empty => `nameAndProductType` necessarily in `labelsByNameAndProductType`

correct?

Reply:
We do the same loop above, and for every target we insert into `labelsByNameAndProductType`. So it minimally has one element for every `target`, yes.